distro_integration.md

```diff
@@ -0,0 +1,405 @@
+your error message or missing functionality into the necessary config options.
+
+## Part 2 - Firmware
+
+Many of the benefits of ARM systems come from their heavy use of co-processors
+for hardware-accelerated tasks like audio processing, compute or video decoding (notably things
+that the x86 world has been trying to catch up with in the last few years).
+
+These co-processors all need to run firmware and that firmware is usually
+tied to your specific device. Some of these are packaged in
+[linux-firmware](https://web.git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/tree/)
+but for most devices there is no clear license and such packaging isn't
+possible. Instead the firmware must be pulled from the Windows partition (and
+dual booting is the only way to get firmware updates). The only "correct" way to
+handle this is to have your users dual-boot and to periodically boot back into
+Windows for updates, before then copying over new firmware versions.
+
+For X1E laptops, the
+[woa-firmware-yoinker](https://github.com/CadmiumLinux/woa-firmware-yoinker/blob/main/yoink-firmware)
+script can be used to fetch this firmware for you after installing. The process
+is fairly straightforward and can be applied to other Snapdragon laptops too,
+however it requires disabling BitLocker encryption on the Windows partition in order to mount the Windows
+partition from Linux.
+
+The best approach today is probably to come up with a manual or semi-manual
+process that users can follow to extract the firmware from windows and put it in
+the right place during or after the installation.
+
+Solving the current licensing issues of firmware distribution is outside
+the scope of this guide.
+
+## Part 3 - Kernel modules and the initramfs
+
+The eternal battle of every device enablement is getting things to work properly
+in the initramfs. This is such a hated process that we have mostly given up and
+just run udevd or even systemd itself in the ramdisk. However one thing we still
+don't do (most of the time anyway) is pull in all the kernel modules.
+
+On typical x86/ACPI devices, the list of modules required for hardware support
+is usually quite generic and mkinitfs implementations have gained the ability
+to automatically figure out what modules are needed for the device you're
+booting on. But - as is the running theme here - ARM devices usually require way
+more modules for basic stuff like display output and keyboard input (the bare
+necessities for full disk encryption!).
+
+The author of this guide recommends that you make your life as simple as
+possible by doing the following:
+
+1. Install **ALL** kernel modules in the initramfs, especially for initial
+   bringup while you're tracking down all the kernel config options.
+2. Once your system is up and running with all the expected functionality
+   working, you can use [the script
+   here](https://gist.github.com/calebccff/f5e33a25c56c1b2bf32c199bdb76e0de) to
```

Comment:
You want to keep the script in the gist or have it together with the doc here?

Reply:
To be fair I don't mind either way. As long as the gist is being kept alive it has the advantage for updating and interaction). Just seen to often that links went into Nirvana after some time.

Reply:
OK, I added a copy of the script to this repo so it's always archived, but kept the gist as the primary link for now, maybe in the future we either just use it from this repo or some other "utils" repo elsewhere...